commands/command_utils.py

```diff
@@ -16,26 +18,52 @@
 LEAPP_UPGRADE_FLAVOUR_SAP_HANA = 'saphana'
 LEAPP_UPGRADE_PATHS = 'upgrade_paths.json'
 
-VERSION_REGEX = re.compile(r"^([1-9]\d*)\.(\d+)$")
 
+_VersionFormat = namedtuple('VersionFormat', ('human_readable', 'regex'))
 
-def check_version(version):
+
+class VersionFormats(Enum):
+    MAJOR_ONLY = _VersionFormat('MAJOR_VER', re.compile(r'^[1-9]\d*$'))
+    MAJOR_MINOR = _VersionFormat('MAJOR_VER.MINOR_VER', re.compile(r"^([1-9]\d*)\.(\d+)$"))
+
+
+class _VersionKind(str, Enum):
+    """ Enum encoding information whether the given OS version is source or target. """
+    SOURCE = 'source'
+    TARGET = 'target'
+
+
+class DistroIDs(str, Enum):
+    RHEL = 'rhel'
+    CENTOS = 'centos'
+
+
+_DISTRO_VERSION_FORMATS = {
+    DistroIDs.RHEL: VersionFormats.MAJOR_MINOR,
+    DistroIDs.CENTOS: VersionFormats.MAJOR_ONLY,
+}
+"""
+Maps distro ID to the expected OS version format.
+
+If a distro is not listed in the dictionary, then VersionFormats.MAJOR_MINOR
+is used as a default.
+"""
```

Comment:
How can consumers extend these lists (without forking the codebase)?
While this probably won't matter much (for Alma and friends) if the fallback is MAJOR_MINOR, but still curious :)

Reply:
Hmm, community projects can always create an upstream PR extending the list, and there should be no problem in merging such a change. Not sure whether implementing something that provides extensibility is worth the effort, since it is not as simple as 'writing an actor' due to the code being executed very early.